scene/2d/tile_map.h

```diff
@@ -166,6 +225,34 @@ class TileMapLayer : public RefCounted {
 		FORMAT_MAX,
 	};
 
+	enum DirtyFlags {
+		DIRTY_FLAGS_LAYER_ENABLED = 0,
+		DIRTY_FLAGS_LAYER_MODULATE,
+		DIRTY_FLAGS_LAYER_Y_SORT_ENABLED,
+		DIRTY_FLAGS_LAYER_Y_SORT_ORIGIN,
+		DIRTY_FLAGS_LAYER_Z_INDEX,
+		DIRTY_FLAGS_LAYER_INDEX_IN_TILE_MAP_NODE,
+		DIRTY_FLAGS_TILE_MAP_IN_TREE = 0,
+		DIRTY_FLAGS_TILE_MAP_IN_CANVAS,
+		DIRTY_FLAGS_TILE_MAP_VISIBILITY,
+		DIRTY_FLAGS_TILE_MAP_XFORM,
+		DIRTY_FLAGS_TILE_MAP_LOCAL_XFORM,
+		DIRTY_FLAGS_TILE_MAP_SELECTED_LAYER,
+		DIRTY_FLAGS_TILE_MAP_LIGHT_MASK,
+		DIRTY_FLAGS_TILE_MAP_MATERIAL,
+		DIRTY_FLAGS_TILE_MAP_USE_PARENT_MATERIAL,
+		DIRTY_FLAGS_TILE_MAP_TEXTURE_FILTER,
+		DIRTY_FLAGS_TILE_MAP_TEXTURE_REPEAT,
+		DIRTY_FLAGS_TILE_MAP_TILE_SET,
+		DIRTY_FLAGS_TILE_MAP_QUADRANT_SIZE,
+		DIRTY_FLAGS_TILE_MAP_COLLISION_ANIMATABLE,
+		DIRTY_FLAGS_TILE_MAP_COLLISION_VISIBILITY_MODE,
+		DIRTY_FLAGS_TILE_MAP_NAVIGATION_VISIBILITY_MODE,
+		DIRTY_FLAGS_TILE_MAP_Y_SORT_ENABLED,
+		DIRTY_FLAGS_TILE_MAP_RUNTIME_UPDATE,
+		DIRTY_FLAGS_MAX,
```

Comment:
Are that many flags actually needed? Some of them seem unused or have duplicate meaning.

Also you are using them with a bool array, but the assigned value resets in the middle (`DIRTY_FLAGS_TILE_MAP_IN_TREE `). This doesn't seem right.

Reply:
Some flags seem unused as they are not, like, used specifically by their name. However they are checked in the `_debug_update` by a loop:
```
	for (int i = 0; i < DIRTY_FLAGS_MAX; i++) {
		if (dirty.flags[i]) {
			anything_changed = true;
			break;
		}
	}
```
I guess I could replace several flags by a single "something changed" flag, but TBH, I prefer to keep it that way in case we need more optimization. It's a bit weird as it means we have a lot of flags, but at least it is consistent and helps when implementing optimizations.